docs/2.8.0/docs/daml-finance/tutorials/advanced-topics/instrument-modeling/date-utilities.rst

```diff
@@ -0,0 +1,279 @@
+  :language: daml
+  :start-after: -- TEST_PREVIOUS_DOM_FROM_EOFEB_BEGIN
+  :end-before: -- TEST_PREVIOUS_DOM_FROM_EOFEB_END
+
+While the above :ref:`RollConvention <module-daml-finance-util-date-rollconvention-88672>` functions
+can be useful on their own, they are often used in the context of rolling out a schedule. This is
+explained in the next section.
+
+Schedule
+========
+
+The :ref:`Schedule <module-daml-finance-util-date-schedule-32303>` module contains functions for
+creating a series of time periods.
+A schedule can be used to represent different aspects of a financial instrument, for example:
+
+- payment periods, which define e.g. *when* a bond coupon or a swap payment should be paid
+- calculation periods, which define e.g. *how* the interest amount is calculated
+
+For example, let us define a periodic 3M schedule that rolls on the 30th of the month:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Schedule.daml
+  :language: daml
+  :start-after: -- CREATE_PERIODIC_SCHEDULE_BEGIN
+  :end-before: -- CREATE_PERIODIC_SCHEDULE_END
+
+Since there is one year between the start and the end date, we would expect this periodic schedule
+to correspond to four periods of 3M each:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Schedule.daml
+  :language: daml
+  :start-after: -- CREATE_EXPECTED_SCHEDULE_RESULT_BEGIN
+  :end-before: -- CREATE_EXPECTED_SCHEDULE_RESULT_END
+
+Note that we distinguish between adjusted and unadjusted dates. The unadjusted date is the result of
+adding a specified time period to the start date of the previous schedule period. This can fall on a
+non-business day. By applying a
+:ref:`BusinessDayAdjustment <type-daml-finance-interface-types-date-calendar-businessdayadjustment-93933>`,
+we also get an adjusted date, which must fall on a business day.
+
+Now, let us roll out this schedule to get the specific start and end date for each period, and
+compare it to the expected result:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Schedule.daml
+  :language: daml
+  :start-after: -- CREATE_SCHEDULE_BEGIN
+  :end-before: -- CREATE_SCHEDULE_END
+
+An important aspect of schedules is the concept of *stub periods*. The above schedule only had
+regular 3M periods. However, if the sample schedule would start one month later:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/Schedule.daml
+  :language: daml
+  :start-after: -- CREATE_PERIODIC_SCHEDULE_WITH_STUB_BEGIN
+  :end-before: -- CREATE_PERIODIC_SCHEDULE_WITH_STUB_END
+
+it would only have 11 months in total, which does not match with neither 3 nor 4 regular 3M periods.
+Instead, we need to define a stub period, so that the rest of the schedule is regular.
+In the example above, we did this by defining a start date for the first regular period. This
+implies a short initial stub period, in our case 2 months. We would expect the following schedule:
```

Comment:
Do you maybe mean "assumes" rather than "implies"? Not sure.

Reply:
I meant that _defining a start date for the first regular period_ has the consequence a short initial stub period will be used, so that would be "implies", right?